bgpd/bgp_route.c

```diff
@@ -8648,12 +8648,16 @@ void bgp_redistribute_add(struct bgp *bgp, struct prefix *p,
 	 */
 	assert(attr.aspath);
 
+	if (p->family == AF_INET6)
+		UNSET_FLAG(attr.flag, ATTR_FLAG_BIT(BGP_ATTR_NEXT_HOP));
```

Comment:
What is the reason for unsetting here?

Reply:
because the NEXTHOP attribute is valid only for ipv4 prefixes.
when creating a BGP update with ipv6, you don't need this flag, and the bgp redistribution was doing this systematically.